packages/trie/src/trie.ts

```diff
@@ -172,6 +172,54 @@ export class Trie {
     return new Trie(opts)
   }
 
+  /**
+   * Static version of fromProof function with the same behavior.
+   * @param proof
+   * @deprecated Use `updateTrieFromProof`
+   */
+  static async fromProof(proof: Proof, opts?: TrieOpts): Promise<Trie> {
+    const trie = await Trie.create(opts)
+    if (opts?.root && !equalsBytes(opts.root, trie.hash(proof[0]))) {
+      throw new Error('Invalid proof provided')
+    }
+    const opStack = proof.map((nodeValue) => {
+      return {
+        type: 'put',
+        key: Uint8Array.from(trie.hash(nodeValue)),
+        value: nodeValue,
+      } as PutBatch
+    })
+    await trie._db.batch(opStack)
+    trie.root(trie.hash(proof[0]))
+    await trie.persistRoot()
+    return trie
```

Comment:
This code is now redundant, can this call into `trie.fromProof`? The `if (opts?.root && !equalsBytes(opts.root, trie.hash(proof[0]))) {` check should stay though.

However, another question then here is, we have `createTrieFromProof` which almost does exactly the same, except that this `fromProof` has the extra check that the `trie.hash(proof[0])` matches the `opts.root` :thinking: (if it is provided :thinking: )

Reply:
I think this one needs to be checked @scorbajio

Reply:
I've switched it over to `updateFromProof` to remove redundancies.